cinema/views.py

```diff
@@ -21,24 +26,41 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(mixins.CreateModelMixin,
+                   mixins.ListModelMixin,
+                   GenericViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
to avoid code duplication add TokenAuthentication to the default authentication_classes in settings.py and add IsAdminOrIfAuthenticatedReadOnly to the default permission_classes in settings.py